File: Flip_bits/code.cpp
class Solution{
    public:
    int maxOnes(int a[], int n)
    {
    int count_zero = 0;
    int count_one = 0;
    int max_zero = INT_MIN;

    for(int i=0; i<n; i++){
        if(a[i] == 0){
            count_zero++;
        }
        else{
            count_zero--;
            count_one++;
        }

        if(count_zero < 0){
            count_zero = 0;
        }
        max_zero = max(max_zero, count_zero);
    }
    return count_one+max_zero;
    }
};

File: Flip_bits/code.java
class Solution {

    public static int maxOnes(int arr[], int n) {
        // Your code goes here
        int total=0,currtotal=0,max=0;
        for(int i=0;i<n;i++){
            if(arr[i]==1) total++;
        }
        max=total;
        for(int i=0;i<n;i++){
            if(arr[i]==1&&currtotal!=0){
                currtotal-=1;
            }else if(arr[i]==0){
                currtotal+=1;
                max=Math.max(max,(currtotal+total));
            }
        }
        return max;
    }
}
